core/src/main/java/org/bitcoinj/crypto/LazyECPoint.java

```diff
@@ -42,15 +42,27 @@ public class LazyECPoint {
     @Nullable
     private ECPoint point;
 
+    /**
+     * Construct a LazyECPoint from a public key. Due to the delayed decoding of the point the validation of the
+     * public key is delayed too, e.g. until a getter is called.
+     *
+     * @param bits  public key bytes
+     */
+    public LazyECPoint(byte[] bits) {
+        this.bits = bits;
+        this.compressed = ECKey.isPubKeyCompressed(bits);
+    }
+
     /**
      * Construct a LazyECPoint from a public key. Due to the delayed decoding of the point the validation of the
      * public key is delayed too, e.g. until a getter is called.
      *
      * @param curve a curve the point is on
      * @param bits  public key bytes
+     * @deprecated Use {@link LazyECPoint#LazyECPoint(byte[])}
      */
+    @Deprecated
     public LazyECPoint(ECCurve curve, byte[] bits) {
-        this.curve = curve;
```

Comment:
I wonder if this can be dangerous for users who specify a different curve than the Bitcoin curve. Not that I expect anyone has ever used this constructor. I would still guard this with an exception.

Reply:
Good idea. I highly doubt anyone is using that with a different curve, though.

Reply:
Resolved and force-pushed.